lib/elixir/test/elixir/kernel_test.exs

```diff
@@ -361,6 +361,18 @@ defmodule KernelTest do
       _ = a
     end
 
+    test "setting attribute with uppercase" do
+      message = "module attributes cannot start with an uppercase letter"
+
+      assert_raise ArgumentError, message, fn ->
+        defmodule InSample do
+          @Upper
+        end
+      end
+    after
+      purge(InSample)
```

Comment:
You don't need to purge it here because the module was never defined in this case. So you can remove the whole `after` bit.